plugins/coverbrowser.koplugin/main.lua

```diff
@@ -0,0 +1,353 @@
+                            callback = function()
+                               self:setupHistoryDisplayMode("list_image_filename")
+                            end,
+                            separator = true,
+                        },
+                    },
+                },
+                {
+                    text = _("Book infos cache maintenance"),
+                    sub_item_table = {
+                        {
+                            text_func = function() -- add current db size to menu text
+                                local BookInfoManager = require("bookinfomanager")
+                                local sstr = BookInfoManager:getDbSize()
+                                return _("Current cache size: ") .. sstr
+                            end,
+                            -- no callback, only for information
+                        },
+                        {
+                            text = _("Purge cache"),
+                            callback = function()
+                                local ConfirmBox = require("ui/widget/confirmbox")
+                                UIManager:close(self.file_dialog)
+                                UIManager:show(ConfirmBox:new{
+                                    text = _("Are you sure that you want to purge cover and metadata cache?\n"),
+                                    ok_text = _("Purge"),
+                                    ok_callback = function()
+                                        local BookInfoManager = require("bookinfomanager")
+                                        BookInfoManager:deleteDb()
+                                    end
+                                })
+                            end,
+                        },
+                    },
+                },
+            },
+        }
+    end
+end
+
+function CoverBrowser:refreshFileManagerInstance(cleanup)
+    local FileManager = require("apps/filemanager/filemanager")
+    local fm = FileManager.instance
+    if fm then
+        local fc = fm.file_chooser
+        if cleanup then -- clean instance properties we may have set
+            if fc.onFileHold_orig then
+                -- remove our onFileHold that extended file_dialog with new buttons
+                fc.onFileHold = fc.onFileHold_orig
+                fc.onFileHold_orig = nil
+                fc.onFileHold_ours = nil
+            end
+        end
+        fc:updateItems()
+    end
+end
+
+function CoverBrowser:setupFileManagerDisplayMode(display_mode)
+    if not display_mode then -- if none provided, use current one
+        display_mode = self.filemanager_display_mode
+    end
+    if not DISPLAY_MODES[display_mode] then
+        display_mode = nil
+    end
+    self.filemanager_display_mode = display_mode
+    G_reader_settings:saveSetting("filemanager_display_mode", self.filemanager_display_mode)
```

Comment:
Incidentally, while I don't exactly perceive it as a problem in this case because I feel that this is perhaps more of a kind of experimental core expansion than a plugin per se, you are effectively polluting the global settings with plugin-specific settings here.

So from my end this is more of a "be aware that this is what you're doing" than a "please change it".